elastic/kibana
# [Infrastructure UI] Improve error handling metric index

## 📓 Summary

Closes #144882 

This PR updates the error handling in case the user configure a remote index that was returning an error, giving more meaningful feedback to the user.

It also includes a refactor for the `useSource` custom hook and removes the usages for the legacy `useSourceViaHttp` custom hook.

## 🧪 Testing

- Navigate to the metrics setting page
- Set a wrongly formatted or non-existing remote cluster
- Verify to get a toast notification about the issue and to get notified about the same issue when accessing inventory and hosts pages.

**Before**
<img width="1741" alt="before" src="https://user-images.githubusercontent.com/34506779/223672635-abb3a7f3-e8a6-43b6-972c-813514d1b63a.png">

**After**

https://user-images.githubusercontent.com/34506779/223672676-017422fb-ccc7-456e-854c-ec79837da281.mp4

## 👣 Next step
As happens in the Data View stack management section, would be nice to inform the user while he types the index pattern if no associated data view matches the typed wildcard.
An additional value could be, in case the user wants to anyway set the index pattern, to inform him with a callout message that it doesn't match any data view, so that the issue cause is immediately clear when no data are viewed on the metrics pages.

- [x] https://github.com/elastic/kibana/issues/152885
